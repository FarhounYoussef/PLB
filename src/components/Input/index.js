import React from 'react';
import {TextInput} from 'react-native';
import Label from '../Label';

import styles from './styles';

class Input extends React.Component {
  constructor(props) {
    super(props);
    this.state = {test: 'value test'};
  }
  render() {
    return (
      <>
        {this.props.label && <Label>{this.props.label}</Label>}
        <TextInput style={styles.input} {...this.props} />
      </>
    );
  }
}

export default Input;
