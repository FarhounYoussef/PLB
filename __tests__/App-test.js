import React from 'react';
import {shallow, configure} from 'enzyme';
import Input from '../src/components/Input';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Testing ReassignLocationMenu component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<Input label={'Test label 1'} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({label: 'Test label 2'});
    expect(wrapper).toMatchSnapshot();
  });
});
