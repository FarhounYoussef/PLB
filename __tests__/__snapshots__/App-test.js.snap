// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ReassignLocationMenu component renders as expected 1`] = `
<Fragment>
  <Label>
    Test label 1
  </Label>
  <TextInput
    label="Test label 1"
    style={
      Object {
        "borderColor": "#bacbde",
        "borderRadius": 10,
        "borderWidth": 1,
        "flex": 1,
        "height": 44,
        "marginRight": 10,
        "paddingLeft": 15,
      }
    }
  />
</Fragment>
`;

exports[`Testing ReassignLocationMenu component renders as expected 2`] = `
<Fragment>
  <Label>
    Test label 2
  </Label>
  <TextInput
    label="Test label 2"
    style={
      Object {
        "borderColor": "#bacbde",
        "borderRadius": 10,
        "borderWidth": 1,
        "flex": 1,
        "height": 44,
        "marginRight": 10,
        "paddingLeft": 15,
      }
    }
  />
</Fragment>
`;
